File: TUnit.TestProject/ConflictingDependsOnTests.cs
using System.Diagnostics.CodeAnalysis;
using TUnit.TestProject.Attributes;

namespace TUnit.TestProject;

[EngineTest(ExpectedResult.Failure)]
[UnconditionalSuppressMessage("Usage", "TUnit0033:Conflicting DependsOn attributes")]
public class ConflictingDependsOnTests
{
    [Test, DependsOn(nameof(Test2))]
    public async Task Test1()
    {
        await Task.Delay(TimeSpan.FromSeconds(5));
    }

    [Test, DependsOn(nameof(Test1))]
    public async Task Test2()
    {
        await Task.CompletedTask;
    }
}

File: TUnit.TestProject.FSharp/ConflictingDependsOnTests.fs
namespace TUnit.TestProject.FSharp

open System
open System.Threading.Tasks
open TUnit.Core

[<UnconditionalSuppressMessage("Usage", "TUnit0033:Conflicting DependsOn attributes")>]
type ConflictingDependsOnTests() =
    [<Test>]
    [<DependsOn("Test2")>]
    member _.Test1() : Task = Task.Delay(TimeSpan.FromSeconds(5.0))

    [<Test>]
    [<DependsOn("Test1")>]
    member _.Test2() : Task = Task.CompletedTask
